src/outlier_hub/datasets/ham10k/factory.py

```diff
@@ -1,75 +1,152 @@
-        dataset_resource = self.storage_connector.get_resource(identifier=dataset_identifier)
+        if not (self.check_exists()):
+            print('Downloading and preparing the dataset.')
+            if split not in self.splits:
+                raise logging.debug(f"Split {split} is not defined.")
+            self._retrieve_raw()
+            self._prepare(split)
+        dataset_identifier = self._get_resource_id(data_type = 'preprocessed', data_split = os.path.join(split, self.dataset_name))
+        print('dataset_identifier:', dataset_identifier)
+        #dataset_resource = self.storage_connector.get_resource(identifier = dataset_identifier)
         meta = MetaFactory.get_iterator_meta(sample_pos=0, target_pos=1, tag_pos=2)
-        return HAMIterator(dataset_resource, split), meta
+        return HAMIterator(dataset_identifier), meta
+                            
 
     def get_dataset_iterator(self, config: Dict[str, Any] = None) -> Tuple[DatasetIteratorIF, IteratorMeta]:
-        if not (self.check_exists()):
-            self._prepare(split="raw")
         return self._get_iterator(**config)
 
-
-# Code for testing the dataset
 if __name__ == "__main__":
+    
     # get root workind directory path
-    root_path = pathlib.Path.cwd().parents[3]
+    root_path = pathlib.Path.cwd()
+    logger.debug(f' pathlib.Path.cwd(): { pathlib.Path.cwd()}')
 
     # complete path to manual added data
     data_path = os.path.join(root_path, "src/outlier_hub/datasets/ham10k/data")
 
     storage_connector = FileStorageConnector(root_path=data_path)
 
-    ham_factory = HAMFactory(storage_connector)
-
-    ham_iterator, _ = ham_factory.get_dataset_iterator(config={"split": "raw"})
-
-    sample, target, tag = ham_iterator[1]
-
-    print(type(ham_iterator[1]))
-    print(type(sample))
-    logger.debug(f"finished")
+    ham10k_factory = Ham10kFactory(storage_connector)
+
+    ham10k_iterator, _ = ham10k_factory.get_dataset_iterator(config={"split": "train"})
+
+    print(len(ham10k_iterator))
+
+    a,b,c,d,e,f,g = 0,0,0,0,0,0,0
+    for i in range(len(ham10k_iterator)):
+        # print('i:',i)
+        sample, target, tag = ham10k_iterator[i]
+        # Image.show(sample)
+        # trans = torchvision.transforms.ToPILImage()
+        # trans1 = torchvision.transforms.ToTensor()
+        # plt.imshow(trans(trans1(sample)))
+        # plt.show()
+        # print(target)
+        # print(tag)
+        # print('target:',target)
+        #print('target:',type(int(target)))
+        #print(int(target) == 1)
+
+        if int(target) == 0:
+            a = a + 1
+        elif int(target) == 1:
+            b = b + 1
+        elif int(target) == 2:
+            c = c + 1
+        elif int(target) == 3:
+            d = d + 1
+        elif int(target) == 4:
+            e = e + 1
+        elif int(target) == 5:
+            f = f + 1
+        elif int(target) == 6:
+            g = g + 1
+    print('verteilung:',a,b,c,d,e,f,g)
```

Comment:
for this we have collections count in python

Reply:
Thanks for the hint - Done